packages/eas-json/src/EasJsonReader.ts

```diff
@@ -23,6 +23,31 @@ interface BuildProfilePreValidation {
 export class EasJsonReader {
   constructor(private projectDir: string, private platform: 'android' | 'ios' | 'all') {}
 
+  private intersect<T>(setA: Set<T>, setB: Set<T>): Set<T> {
+    return new Set([...setA].filter(i => setB.has(i)));
+  }
+
+  /**
+   * Return build profile names for a particular platform.
+   * If platform is 'all', return common build profiles for all platforms
+   */
+  public async getBuildProfileNamesAsync(): Promise<string[]> {
+    const easJson = await this.readRawAsync();
+    let intersectingProfileNames: Set<string> | null = null;
+    for (const platform of ['android', 'ios'] as ('android' | 'ios')[]) {
+      const profiles = (easJson.builds ? easJson.builds[platform] : {}) ?? {};
+      const profileNames = new Set(Object.keys(profiles));
+      if (this.platform === platform) {
+        return Array.from(profileNames);
+      } else if (this.platform === 'all') {
+        intersectingProfileNames = intersectingProfileNames
+          ? this.intersect(intersectingProfileNames, profileNames)
+          : profileNames;
+      }
+    }
+    return intersectingProfileNames ? Array.from(intersectingProfileNames) : [];
```

Comment:
```suggestion
    if (this.platform === 'android') {
      return Object.keys(easJson?.builds?.android ?? {});
    } else if (this.platform === 'ios') {
      return Object.keys(easJson?.builds?.ios ?? {});
    } else {
       const intersectingProfileNames = this.intersect(
         Object.keys(easJson?.builds?.ios ?? {}),
         Object.keys(easJson?.builds?.android ?? {})
       );
       return Array.from(intersectingProfileNames)
    }
```
It's correct behavior, but it's hard to read it